python/lsst/ap/association/trailedSourceFilter.py

```diff
@@ -0,0 +1,110 @@
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.utils.timer import timeMethod
+
+# import numpy as np
+import pandas as pd
+
+# Enforce an error for unsafe column/array value setting in pandas.
+pd.options.mode.chained_assignment = 'raise'
+
+
+class TrailedSourceFilterConfig(pexConfig.Config):
+    """Config class for TrailedSourceFilterTask.
+    """
+    maxTrailLength = pexConfig.Field(
+        dtype=float,
+        doc='Maximum trail length permitted is less than 10 degrees/day. This is a rate '
+            'of 0.416 arcseconds per second.As trail length is measured in '
+            'arcseconds, it is dependant on the length of the exposure.',
+        default=0.416,  # HSC Default. Should Decam and LSST defaults be passed? Does it change possibly?
+    )
+
+
+class TrailedSourceFilterTask(pipeBase.Task):
+    """Find trailed sources in DIAObjects.
+
+    """
+
+    ConfigClass = TrailedSourceFilterConfig
+    _DefaultName = "trailedAssociation"
+
+    @timeMethod
+    def run(self,
+            dia_sources, exposure):
+        """Find trailed sources which have not been filtered out and will
+        not be included in the diaSource catalog.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            New DIASources to be checked for trailed sources.
+        exposure : `pandas.DataFrame`
+            Calibrated exposure differenced with a template image during
+            image differencing.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results struct with components.
+
+            - ``"dia_sources"`` : DiaSource table that is free from unwanted
+             trailed sources (`pandas.DataFrame`)
+
+            - ``"trailed_dia_sources"`` : DiaSources that have trailed more than
+             0.416 arcseconds/second*exposure_time(`pandas.DataFrame`)
+        """
+        trail_mask = self.check_dia_source_trail(dia_sources, exposure)
+
+        return pipeBase.Struct(
+            diaSources=dia_sources[~trail_mask].reset_index(drop=True),
+            trailedDiaSources=dia_sources[trail_mask].reset_index(drop=True))
+
+    def check_dia_source_trail(self, dia_sources, exposure):
+        """Check that all DiaSources have trails.
+
+        Creates a mask for sources with lengths greater than 0.416 arcseconds/second*exposure time.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            Input DiaSources to check for trail lengths.
+        exposure : `pandas.DataFrame`
+            Calibrated exposure differenced with a template image during
+            image differencing.
+
+        Returns
+        -------
+        trail_mask : `pandas.DataFrame`
+            Boolean mask for dia_sources which are greater than the cuttoff length.
+        """
+        exposure_time = exposure.getInfo().getVisitInfo().getExposureTime()
```

Comment:
The only thing you use the exposure for here is the exposure time? In that case, you should just be passing in the time (to this method, and `run`), as you don't need the rest of the exposure at all. There's no pixel math happening here, so all we need is the time (and note in the docstrings that it's in seconds).

Reply:
Why do you need the full exposure, and not just the exposure time here?